packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -0,0 +1,782 @@
+  currentUser: UsersCurrent; // replace with the correct type
+  loadingNewDialogue: boolean;
+  createDialogue: (title: string, participants: string[]) => void;
+  classes: ClassesType;
+};
+
+const MatchDialogueButton: React.FC<MatchDialogueButtonProps> = ({
+  isMatched,
+  targetUserId,
+  targetUserDisplayName,
+  currentUser,
+  loadingNewDialogue,
+  createDialogue,
+  classes,
+}) => {
+  if (!isMatched) return <div></div>; // need this instead of null to keep the table columns aligned
+
+  return (
+    <div>
+      <button
+        className={classes.newDialogueButton}
+        onClick={(e) =>
+          createDialogue(
+            `${currentUser?.displayName}/${targetUserDisplayName}`,
+            [targetUserId]
+          )
+        }
+      >
+        {loadingNewDialogue ? <a data-cy="message">Creating New Dialogue...</a> : <a data-cy="message">Start Dialogue</a>}
+      </button>
+    </div>
+  );
+};
+
+const MessageButton: React.FC<{
+  targetUserId: string;
+  currentUser: UsersCurrent; // replace with the correct type
+  classes: ClassesType;
+}> = ({ targetUserId, currentUser, classes }) => {
+  const { NewConversationButton } = Components;
+  
+  return (
+    <button className={classes.messageButton}>
+      <NewConversationButton user={{_id: targetUserId}} currentUser={currentUser}>
+        <a data-cy="message">Message</a>
+      </NewConversationButton>
+    </button>
+  );
+};
+
+export const DialogueMatchingPage = ({classes}: {
+  classes: ClassesType,
+}) => {
+  
+  const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
+
+  const updateCurrentUser = useUpdateCurrentUser()
+  const [optIn, setOptIn] = React.useState(false); // for rendering the checkbox
+
+  const { UsersName, Loading, LoadMore } = Components;
+
+  const {create: createPost, loading: loadingNewDialogue, error: newDialogueError} = useCreate({ collectionName: "Posts", fragmentName: "PostsEdit" });
+  const { history } = useNavigation();
+
+  const { loading, error, data } = useQuery(gql`
+    query getDialogueUsers {
+      GetUserDialogueUsefulData {
+        dialogueUsers {
+          _id
+          displayName
+          karma
+        }
+        topUsers {
+          _id
+          displayName
+          total_power
+          total_agreement
+        }
+       }
+    }
+  `);
+
+  const userDialogueUsefulData:UserDialogueUsefulData = data?.GetUserDialogueUsefulData
+  const currentUser = useCurrentUser();
+
+  const {loading: userLoading, results: userDialogueChecks} = useMulti({
+    terms: {
+      view: "userDialogueChecks",
+      userId: currentUser?._id,
+      limit: 1000,
+    },
+    fragmentName: "DialogueCheckInfo",
+    collectionName: "DialogueChecks",
+  });
+
+  const {loading: userOptedInLoading, results : UsersOptedInToDialogueFacilitation, loadMoreProps} = useMulti({
+    terms: { 
+      view: 'usersWithOptedInToDialogueFacilitation',
+      limit: 10, 
+      sort: { karma: -1 } 
```

Comment:
I don't think this `sort` is getting used at all when passed in as a term, since it's defined inline in the `usersWithOptedInToDialogueFacilitation` view function itself.  If it is, it's redundant.